Wikipedia/Code/EditSaveViewController.swift

```diff
@@ -348,6 +349,9 @@ class EditSaveViewController: WMFScrollViewController, Themeable, UITextFieldDel
         default:
             WMFAlertManager.sharedInstance.showErrorAlert(nsError, sticky: true, dismissPreviousAlerts: true, tapCallBack: nil)
         }
+        if let articleURL {
+            EditAttemptFunnel.shared.logSaveFailure(articleURL: articleURL)
+        }
```

Comment:
Would it make sense to move this block before the `switch` statement to ensure it's called regardless if one of the `switch` cases returns early?

Reply:
Yes! Thanks for catching all those :) I guess it's good for another review, let me know if I can adjust something else